src/status_im/contexts/onboarding/create_password/view.cljs

```diff
@@ -29,78 +27,91 @@
                           :status :error
                           :shown? true}
                          {:text   (i18n/label :t/password-creation-hint)
                           :status hint-1-status
                           :shown? true})
        :placeholder    (i18n/label :t/password-creation-placeholder-1)
        :on-change-text on-change-password
-       :on-focus       on-input-focus
        :auto-focus     true}]
      [rn/view {:style style/space-between-inputs}]
      [password-with-hint/view
       {:hint           {:text   hint-2-text
                         :status hint-2-status
-                        :shown? (and (not empty-password?)
-                                     show-password-validation?)}
+                        :shown? (and non-empty-password? show-validation?)}
        :error?         error?
        :placeholder    (i18n/label :t/password-creation-placeholder-2)
        :on-change-text on-change-repeat-password
-       :on-focus       on-input-focus
        :on-blur        on-blur-repeat-password}]]))
 
 (defn help
-  [{{:keys [lower-case? upper-case? numbers? symbols?]} :validations
-    password-strength                                   :password-strength}]
-  [rn/view
-   [quo/strength-divider {:type (constants/strength-status password-strength :info)}
-    (i18n/label :t/password-creation-tips-title)]
-   [rn/view {:style style/password-tips}
-    [quo/tips {:completed? lower-case?}
-     (i18n/label :t/password-creation-tips-1)]
-    [quo/tips {:completed? upper-case?}
-     (i18n/label :t/password-creation-tips-2)]
-    [quo/tips {:completed? numbers?}
-     (i18n/label :t/password-creation-tips-3)]
-    [quo/tips {:completed? symbols?}
-     (i18n/label :t/password-creation-tips-4)]]])
+  [{:keys [lower-case? upper-case? numbers? symbols?] :as validations}]
+  (let [password-strength (constants/strength-status (password/strength validations) :info)]
+    [rn/view
+     [quo/strength-divider {:type password-strength}
+      (i18n/label :t/password-creation-tips-title)]
+     [rn/view {:style style/password-tips}
+      [quo/tips {:completed? lower-case?}
+       (i18n/label :t/password-creation-tips-1)]
+      [quo/tips {:completed? upper-case?}
+       (i18n/label :t/password-creation-tips-2)]
+      [quo/tips {:completed? numbers?}
+       (i18n/label :t/password-creation-tips-3)]
+      [quo/tips {:completed? symbols?}
+       (i18n/label :t/password-creation-tips-4)]]]))
 
 (defn- use-password-checks
   [password]
   (rn/use-memo
    (fn []
-     (let [{:keys [long-enough? short-enough?]
-            :as   validations} (password/validate password)]
-       {:password-long-enough?  long-enough?
-        :password-short-enough? short-enough?
-        :password-validations   validations
-        :password-strength      (password/strength validations)
-        :empty-password?        (empty? password)}))
+     (-> password
+         (password/validate)
+         (assoc :non-empty? (seq password))))
```

Comment:
Simplified the data returned.

We  duplicated the data in the previous destructuring